src/cryptography/hazmat/bindings/openssl/binding.py

```diff
@@ -172,3 +172,11 @@ def _lock_cb(cls, mode, n, file, line):
                     mode, n, file, line
                 )
             )
+
+
+# OpenSSL is not thread safe until the locks are initialized. We call this
+# method in module scope so that it executes with the import lock. On
+# Pythons < 3.4 this import lock is a global lock, which can prevent a race
+# condition registering the OpenSSL locks. On Python 3.4+ the import lock
+# is per module so this approach will not work.
```

Comment:
It didn't seem to work for people who were testing, but it won't hurt so maybe we should still do it?